source/slang/slang-emit-spirv.cpp

```diff
@@ -1791,6 +1791,26 @@ struct SPIRVEmitContext
         }
     }
 
+    static SpvStorageClass getSpvStorageClass(IRPtrTypeBase* ptrType)
+    {
+        SpvStorageClass storageClass = SpvStorageClassFunction;
+        if (ptrType && ptrType->hasAddressSpace())
+        {
+            storageClass = (SpvStorageClass)ptrType->getAddressSpace();
+        }
+        return storageClass;
+    }
+
+    // https://registry.khronos.org/vulkan/specs/1.3/html/chap37.html#VUID-StandaloneSpirv-DescriptorSet-06491
+    // Only UniformConstant, Uniform or StorageBuffer storage class are allowed to be decorated with descriptor
+    // set or binding.
+    static inline bool isBindingAllowed(SpvStorageClass storageClass)
+    {
+        return  (storageClass == SpvStorageClassUniformConstant) ||
```

Comment:
Use `switch` to list all cases.

Reply:
You want me to list all storage classes or just these 3? There are lots of cases.

Reply:
Just turn this into a switch statement and list the three cases. No need to list everything.

Reply:
OK, I see.
But why we prefer the switch..case to this logic expression? Is just a performance concern?